src/ios/Calendar.m

```diff
@@ -501,8 +501,10 @@ - (void) openCalendar:(CDVInvokedUrlCommand*)command {
       
     NSInteger interval = [openDate timeIntervalSinceReferenceDate];
       
-    NSURL *url = [NSURL URLWithString:[NSString stringWithFormat:@"calshow:%ld", interval]];
-    [[UIApplication sharedApplication] openURL:url];
+    dispatch_async(dispatch_get_main_queue(), ^{
+        NSURL *url = [NSURL URLWithString:[NSString stringWithFormat:@"calshow:%ld", interval]];
```

Comment:
Does this have to live inside the main queue?

Reply:
It's not strictly necessary; it’s more about maintaining context and avoiding having just a single line in the block. But I can remove it if you'd like.

Reply:
I was thinking more in terms of if we're maintaining context, then `interval` should also be there (as it's only used by the set `url` call), and then the same for `openDate`and so on... which would result in not needing to have the `runInBackground` call.

Reply:
I can update to something like this, for example:

```
- (void) openCalendar:(CDVInvokedUrlCommand*)command {
  NSDictionary* options = [command.arguments objectAtIndex:0];
  NSNumber* date = [options objectForKey:@"date"];

  dispatch_async(dispatch_get_main_queue(), ^{
    NSDate *openDate;
    NSTimeInterval _startInterval = [date doubleValue] / 1000; // strip millis
      
    //date comes with negative value, corresponds to either empty date field or dates below 1970, use current date
    if ([date doubleValue] < 0){
      openDate = [NSDate date];
    }
    else if ([date doubleValue] > 0){
     openDate = [NSDate dateWithTimeIntervalSince1970:_startInterval];
    }
      
    NSInteger interval = [openDate timeIntervalSinceReferenceDate];
    NSURL *url = [NSURL URLWithString:[NSString stringWithFormat:@"calshow:%ld", interval]];
    [[UIApplication sharedApplication] openURL:url options:@{} completionHandler:nil];
   });
}
```

Reply:
Similar to the background code approach or also include the first two lines of code inside the block.

Reply:
IIRC, calling plugin methods from a background thread is generally not recommended, so we could simply remove the ` dispatch_async(dispatch_get_main_queue(), ^{` altogether. 
However, we can safeguard things on our side and leave it as you wrote in the last comment (if you wish, you can also move `options` and `date` as they are also "UI-related" in this context).

Reply:
Great, made a new update according to the comment.